python/pyarrow/tests/test_dataset.py

```diff
@@ -391,6 +393,14 @@ def test_dataset(dataset, dataset_reader):
     assert sorted(result['group']) == [1, 2]
     assert sorted(result['key']) == ['xxx', 'yyy']
 
+    condition = ds.field(('struct', 'b')) == '1'
+    result = dataset.to_table(use_threads=True, filter=condition).to_pydict()
```

Comment:
Can we test a column projection as well somewhere here?

Reply:
Sure, will do 👍